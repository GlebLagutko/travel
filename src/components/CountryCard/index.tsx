import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import {Link} from 'react-router-dom';
import {useSelector, useDispatch} from "react-redux";
import {CHANGE_COUNTRY} from "../../store/actions/ChangeCountry";
import {saveState} from "../../store/SaveState";
import store from "../../store";

const useStyles = makeStyles({
    root: {
        height: 300,
        width: 250
    },
    media: {
        height: 200,
    },
});


export default function CountryCard({country}) {
    const classes = useStyles();

    const dispatch = useDispatch();

    return (
        <Card className={classes.root} onClick={() => {
            dispatch({
                type: CHANGE_COUNTRY,
                value: {urlName: country.urlName, name: country.urlName[0].toUpperCase() + country.urlName.substr(2)},
            });
            saveState(store);
        }}>
            <CardActionArea>
                <CardMedia
                    className={classes.media}
                    image={`assets/images/thumbnail-${country.urlName}.jpg`}
                    title="Contemplative Reptile"
                />
                <CardContent>
                    <Typography gutterBottom variant="h5" component="h2">
                        {country.name}
                    </Typography>
                    <Typography gutterBottom variant="h5" component="h2">
                        {country.capital}
                    </Typography>
                </CardContent>
            </CardActionArea>
        </Card>
    );
}
