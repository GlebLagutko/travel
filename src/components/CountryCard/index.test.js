import {create} from 'react-test-renderer';

import React from "react";
import CountryCard from ".";
import configureStore from 'redux-mock-store'
import {Provider} from 'react-redux'

describe('CountryCard', () => {
  it('renders correctly', async () => {

    const initialState = {value: {language: "en"}};
    const mockStore = configureStore();
    let store = mockStore(initialState);

    let country = {
        "id": 1,
        "name": "Norway",
        "video": "https://www.youtube.com/embed/Scxs7L0vhZ4",
        "urlName": "norway",
        "capital": "Oslo",
        "description": "Norway, country of northern Europe that occupies the westernhalf of the Scandinavian peninsula. Nearly half of the inhabitants of the country live in the far south, in the region around Oslo, the capital. About two-thirds of Norway is mountainous, and off its much-indented coastline lie, carved by deep glacial fjords, some 50,000 islands.",
        "currency": "NOK",
        "map": "mapbox://styles/gleblagutko1/ckm3hgeoeb04a17qp1zl3qhx6",
        "image": "assets/images/norway.jpg",
        "coordinates": {
          "lon": 10.757933,
          "lat": 59.911491
        },
        "Attractions": [
          {
            "id": 19,
            "name": "Sognefjord",
            "description": "The largest of Norway's fjords, Sognefjord reaches 204 kilometers inland from the coastal village of Skjolden and branches off into countless smaller inlets and fjords along the way.",
            "image": "/assets/images/norway/sognefjord.jpg",
            "Votes": []
          }
        ]
      }
    ;

    const testRenderer = create(
      <Provider store={store}><CountryCard country={country} key={country.id}/></Provider>
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();

  });

});