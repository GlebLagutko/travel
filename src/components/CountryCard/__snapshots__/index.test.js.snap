// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryCard renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  onClick={[Function]}
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardMedia-root makeStyles-media-2"
      style={
        Object {
          "backgroundImage": "url(\\"assets/images/thumbnail-norway.jpg\\")",
        }
      }
      title="Contemplative Reptile"
    />
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        <span>
          Norway
        </span>
      </h2>
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        <span>
          Oslo
        </span>
      </h2>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
